sdk/java/runtime/template/src/main/java/io/dagger/sample/module/DaggerModule.java

```diff
@@ -0,0 +1,38 @@
+package io.dagger.sample.module;
+
+import io.dagger.client.Container;
+import io.dagger.client.DaggerQueryException;
+import io.dagger.client.Directory;
+import io.dagger.module.Base;
+import io.dagger.module.annotation.Function;
+import io.dagger.module.annotation.Object;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+/** DaggerModule main object */
+@Object(value = "dagger-module-name")
+public class DaggerModule extends Base {
+  public DaggerModule() {
+    super();
+  }
+
+  /** Returns a container that echoes whatever string argument is provided */
+  @Function
+  public Container containerEcho(String stringArg) {
+    return dag.container()
```

Comment:
Curious, where does `dag` come from?

Reply:
`dag` is defined in the superclass `Base`. Maybe we could find a better name like `AbstractModule`

Reply:
Just got surprised about the implicit reference. 🙂 I expected a `this.dag` if it's an object instance parameter.

Reply:
In java `this` is not mandatory :-)
I can add it in the samples so that's more explicit if we think that's better.

Reply:
I like without it because it's more consistent with the other SDKs. 🙂